kpi/deployment_backends/kobocat_backend.py

```diff
@@ -886,3 +1005,58 @@ def __prepare_as_drf_response_signature(requests_response):
 
         return prepared_drf_response
 
+    @staticmethod
+    def __validate_payload(payload: dict) -> dict:
+        """
+        Validating the request payload for bulk updating of submissions
+        """
+        if not 'submission_ids' in payload:
+            raise KobocatBulkUpdateSubmissionsClientException(
+                detail=_('`submission_ids` must be included in the payload')
+            )
+
+        if not isinstance(payload['submission_ids'], list):
+            raise KobocatBulkUpdateSubmissionsClientException(
+                detail=_('`submission_ids` must be an array')
+            )
+
+        if len(payload) < 2:
+            raise KobocatBulkUpdateSubmissionsClientException(
+                detail=_('Payload must contain data to update the submissions')
+            )
+
+        return payload
+
+    @classmethod
+    def __prepare_payload(cls, request_data: dict) -> dict:
+        """
+        Preparing the request payload for bulk updating of submissions
+        """
+        payload = json.loads(request_data['payload'][0])
+        validated_payload = cls.__validate_payload(payload)
+
+        # Ensuring submission ids are integer values and unique
+        validated_payload['submission_ids'] = list(
+            set(map(int, validated_payload['submission_ids']))
+        )
+
+        # Sanitizing the payload of potentially destructive keys
+        santized_payload = copy.copy(validated_payload)
+        for key in validated_payload:
+            if key == 'submission_ids':
```

Comment:
aha, so if there were a question named `submission_ids`, we wouldn't be able to bulk update its responses? could the request format be restructured to avoid the possibility of this collision?

Reply:
Because the payload is structured like this (which is the same as `/validation_statuses/`):

```
{
    "submission_ids": [],
    "foo": "bar",
    ...
}
```
I just make sure to pass over `submission_ids` before popping keys off. But you've got a good point about restructuring the payload so this isn't necessary. Something like this might be better?

```
{
    "submission_ids": [],
    "data": {
        "foo": "bar",
        ...
    }
}
```